halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/SecurityConfig.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.config.v1;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@Configuration
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
```

Comment:
Was there an implicit permit all configuration in boot1 that now needs to be explicit?

Reply:
It wasn't implicit in boot1:
* https://github.com/spinnaker/halyard/blob/master/halyard-web/config/halyard.yml#L42
* https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide#actuator-security

Now by default the actuator endpoints in boot2 would be protected by basic auth with the randomly generated password (using boot's autoconfigured user details services), unless at least one user-provided WebSecurityConfigurerAdapter is detected:
* https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide#security

This is our case: `ActuatorEndpointsConfiguration` and `MetricsEndpointConfiguration`. They both permit requests to some endpoints but specify nothing else, leaving everything open. Which is fine but is kinda implicit and almost happening accidently as we include these security configs component by scanning the kork classes.

Hence I proposed to make the configuration explicit by excluding those beans:
* https://github.com/pdelagrave/halyard/blob/boot2/halyard-web/src/main/java/com/netflix/spinnaker/halyard/Main.java#L42

and by adding this `SecurityConfig` class.

Reply:
Sounds good, that makes perfect sense---thanks for the explanation!